core/dictgen/src/TModuleGenerator.cxx

```diff
@@ -380,6 +380,11 @@ void TModuleGenerator::WriteRegistrationSourceImpl(std::ostream& out,
           "    DictInit() {\n"
           "      TriggerDictionaryInitialization_" << dictName << "_Impl();\n"
           "    }\n"
+          "    ~DictInit() {\n"
+          "        assert (isInitialized);"
+          "        TROOT::UnRegisterModule(\"" << demangledDictName << "\","
+                                           "TriggerDictionaryInitialization_" << dictName << "_Impl);\n"
```

Comment:
I suppose that this is misleading (albeit harmeless) as you would a want a ```UnTrigger``` function here.

Reply:
I use that function to find back the shared object. We can avoid that by keeping a mapping in TCling between `modulename` and shared object. Then this function will take only one argument.

Reply:
fair enough.  thanks.

Reply:
Shall I make that change?

Reply:
I would prefer that (but feel free not too if you prefer to leave it as is).

Reply:
@pcanal, could you provide an implementation of `TCling::UnLoadPCM`?